FORMAT_ADIOS.md

```diff
@@ -32,3 +32,35 @@ Output from `bpls -A` for a boolean attribute `pybool` stored in the location of
 
 There is no convention yet for a unique representation of ADIOS2 variables with boolean type.
 Thus, implementations should cast the data to and from `unsigned char` instead.
+
+## `stepBased` Encoding of Iterations
+
+In order to correlate openPMD iterations with ADIOS steps, the *root* group (path `/`) in ADIOS must contain a variable:
+
+  - `__step__`
```

Comment:
Make sure we used `__step__` or `__steps__`

Reply:
Discussed with @franzpoeschel let's call this `snapshot` without `__` and with openPMD 2.0 "iteration" naming: https://github.com/openPMD/openPMD-standard/issues/148

Reply:
Note that in the current implementations this attribute is not part of the root group, but in `/data`, so it's `/data/snapshot`. 
In variable-based encoding this makes the attribute look part of the iteration:

```
> bpls ../samples/variableBasedSeries.bp4 -alt                                                                    
Step 0:                                                                                                                      
  string    /basePath                                                            attr   = "/data/%T/"                        
  uint64_t  /data/changing_value                                                 attr   = 0                                  
  double    /data/dt                                                             attr   = 1                                  
  double    /data/meshes/E/0/position                                            attr   = 0                                  
  uint64_t  /data/meshes/E/0/shape                                               attr   = 1                                  
  double    /data/meshes/E/0/unitSI                                              attr   = 1                                  
  uint64_t  /data/meshes/E/0/value                                               attr   = 0                                  
  uint64_t  /data/meshes/E/attr_0                                                attr   = 0                                  
  string    /data/meshes/E/axisLabels                                            attr   = {"x"}                              
  string    /data/meshes/E/dataOrder                                             attr   = "C"                                
  string    /data/meshes/E/geometry                                              attr   = "cartesian"                        
  double    /data/meshes/E/gridGlobalOffset                                      attr   = 0                                  
  double    /data/meshes/E/gridSpacing                                           attr   = 1                                  
  double    /data/meshes/E/gridUnitSI                                            attr   = 1                                  
  float     /data/meshes/E/timeOffset                                            attr   = 0                                  
  double    /data/meshes/E/unitDimension                                         attr   = {0, 0, 0, 0, 0, 0, 0}              
  int32_t   /data/meshes/E/x                                                     {1000}                                      
  double    /data/meshes/E/x/position                                            attr   = 0
  double    /data/meshes/E/x/unitSI                                              attr   = 1
  int32_t   /data/meshes/E/y                                                     {1}
  double    /data/meshes/E/y/position                                            attr   = 0
  double    /data/meshes/E/y/unitSI                                              attr   = 1
  string    /data/meshes/changing_constant/axisLabels                            attr   = {"x"}
  uint64_t  /data/meshes/changing_constant/changing_constant/shape               attr   = 0
  uint64_t  /data/meshes/changing_constant/changing_constant/value               attr   = 0
  string    /data/meshes/changing_constant/dataOrder                             attr   = "C"
  string    /data/meshes/changing_constant/geometry                              attr   = "cartesian"
  double    /data/meshes/changing_constant/gridGlobalOffset                      attr   = 0
  double    /data/meshes/changing_constant/gridSpacing                           attr   = 1
  double    /data/meshes/changing_constant/gridUnitSI                            attr   = 1
  double    /data/meshes/changing_constant/position                              attr   = 0
  uint64_t  /data/meshes/changing_constant/shape                                 attr   = 0
  float     /data/meshes/changing_constant/timeOffset                            attr   = 0
  double    /data/meshes/changing_constant/unitDimension                         attr   = {0, 0, 0, 0, 0, 0, 0}
  double    /data/meshes/changing_constant/unitSI                                attr   = 1
  uint64_t  /data/meshes/changing_constant/value                                 attr   = 0
  uint64_t  /data/particles/changing_constant/position/position/shape            attr   = 0
  uint64_t  /data/particles/changing_constant/position/position/value            attr   = 0
  uint64_t  /data/particles/changing_constant/position/shape                     attr   = 0
  float     /data/particles/changing_constant/position/timeOffset                attr   = 0
  double    /data/particles/changing_constant/position/unitDimension             attr   = {1, 0, 0, 0, 0, 0, 0}
  double    /data/particles/changing_constant/position/unitSI                    attr   = 1
  uint64_t  /data/particles/changing_constant/position/value                     attr   = 0
  uint64_t  /data/snapshot                                                       attr   = 0
  double    /data/time                                                           attr   = 0
  double    /data/timeUnitSI                                                     attr   = 1
  string    /date                                                                attr   = "2022-08-19 08:35:01 +0000"
  string    /iterationEncoding                                                   attr   = "variableBased"
  string    /iterationFormat                                                     attr   = "/data"
  string    /meshesPath                                                          attr   = "meshes/"
  string    /openPMD                                                             attr   = "1.1.0"
  uint32_t  /openPMDextension                                                    attr   = 0
  string    /particlesPath                                                       attr   = "particles/"
  string    /software                                                            attr   = "openPMD-api"
  string    /softwareVersion                                                     attr   = "0.15.0-dev"
```

In group-based encoding however, the `snapshot` attribute is then at the level above the single iterations:
```
> bpls ../samples/bp4steps_yes_yes.bp/ -alt                                                                       
Step 0:                                                                                                                      
  string    /basePath                                  attr   = "/data/%T/"                                                  
  double    /data/0/dt                                 attr   = 1                                                            
  string    /data/0/meshes/E/axisLabels                attr   = {"x"}                                                        
  string    /data/0/meshes/E/dataOrder                 attr   = "C"                                                          
  string    /data/0/meshes/E/geometry                  attr   = "cartesian"                                                  
  double    /data/0/meshes/E/gridGlobalOffset          attr   = 0                                                            
  double    /data/0/meshes/E/gridSpacing               attr   = 1                                                            
  double    /data/0/meshes/E/gridUnitSI                attr   = 1                                                            
  float     /data/0/meshes/E/timeOffset                attr   = 0                                                            
  double    /data/0/meshes/E/unitDimension             attr   = {0, 0, 0, 0, 0, 0, 0}                                        
  string    /data/0/meshes/E/vector_of_string          attr   = {"vector", "of", "string"}                                   
  int32_t   /data/0/meshes/E/x                         {10}                                                                  
  double    /data/0/meshes/E/x/position                attr   = 0                                                            
  double    /data/0/meshes/E/x/unitSI                  attr   = 1                                                            
  double    /data/0/time                               attr   = 0                                                            
  double    /data/0/timeUnitSI                         attr   = 1                                                            
  uint64_t  /data/snapshot                             attr   = 0                                                            
  string    /date                                      attr   = "2022-08-19 08:38:04 +0000"                                  
  string    /iterationEncoding                         attr   = "groupBased"                                                 
  string    /iterationFormat                           attr   = "/data/%T/"                                                  
  string    /meshesPath                                attr   = "meshes/"
  string    /openPMD                                   attr   = "1.1.0"
  uint32_t  /openPMDextension                          attr   = 0
  string    /software                                  attr   = "openPMD-api"
  string    /softwareVersion                           attr   = "0.15.0-dev"
```